dill/session.py

```diff
@@ -333,12 +428,66 @@ def load_module(
 
     Raises:
         :py:exc:`UnpicklingError`: if unpickling fails.
-        :py:exc:`ValueError`: if the argument ``main`` and module saved
+        :py:exc:`ValueError`: if the argument ``module`` and module saved
             at ``filename`` are incompatible.
 
     Returns:
         A module object, if the saved module is not :py:mod:`__main__` or
-        a module instance wasn't provided with the argument ``main``.
+        a module instance wasn't provided with the argument ``module``.
+
+    Passing an argument to ``module`` forces `dill` to verify that the module
+    being loaded is compatible with the argument value.  Additionally, if the
+    argument is a module (instead of a module name), it supresses the return
+    value. Each case and behavior is exemplified below:
```

Comment:
Thanks for detailed case-by-case examples. I'm going to have to come back to this.